### PR TITLE
Fix superclass lazy relationships not loading properly on subclass

### PR DESCRIPTION
**Pull request by willcosgrove**
Hey @Bajena, I didn't know the best way to test this, and I unfortunately didn't have much time at the moment to try and work out the best way to test it 😞 But all the existing tests pass, so at the worst, it didn't break any existing functionality 😄

### CODE REVIEW

**Comment by Bajena** [comment]
Hey, I tried reproducing your problem in a test case: https://github.com/Bajena/ams_lazy_relationships/pull/18 but the tests seem to pass even without applying your trick. Do you think I might be doing sometihng wrong?

**Comment by willcosgrove** [comment]
The issue with the test is that it is only testing that the relationship is present on the subclass, which it is because inheritance is working properly in AMS relationships.

But you'll find that the `@lazy_relationships` class instance variable on the subclass contains no relationships, which means that none of the relationships will be preloaded when `load_all_lazy_relationships` is called.

So the problem is not that the relationship doesn't exist (which is what the tests you wrote currently check for) but that the relationships don't get preloaded when `load_all_lazy_relationships` is called. I could see how testing that might be kind of complex.  I wasn't able to think of a quick way to test it properly off the top of my head.  But if you add

```rb
lazy_relationships = level0_serializer_class.instance_variable_get(:@lazy_relationships)
expect(lazy_relationships).to be_a(Hash)
expect(lazy_relationships.size).to eq(2)
```

to your test on line 511, you should see it fail when my patch is commented out, and pass when it's active.  Now, that isn't a great test, since it's checking for the exact thing my code does.  It would be best to test it at a higher level that batch loading is happening.  I unfortunately don't have the time to figure it out at the moment 😞 I'm sorry

File: lib/ams_lazy_relationships/core.rb [hunk 1]
```diff
@@ -28,6 +28,11 @@ module ClassMethods
     include RelationshipWrapperMethods
     include Evaluation
 
+    def inherited(subclass)
+      super
+      subclass.instance_variable_set(:@lazy_relationships, @lazy_relationships.clone)
```
Comment (Bajena) [inline lib/ams_lazy_relationships/core.rb]: I got this error while running tests:
```
  1) AmsLazyRelationships::Core inheritance of lazy relationships copies relationships to inherited serializer
     Failure/Error: :@lazy_relationships, @lazy_relationships.clone

     TypeError:
       can't clone NilClass
```

This fixes the problem:
```ruby
      if @lazy_relationships
        subclass.instance_variable_set(
          :@lazy_relationships, @lazy_relationships.clone
        )
      end
```

**Comment by Bajena** [comment]
Ah, also you'll need to rebase to master

**Comment by willcosgrove** [comment]
@Bajena Good catch with that nil bug!  Just added the conditional and rebased, should be good to go!

**Pull request merged**